Floofbot/Modules/ModMail.cs

```diff
@@ -0,0 +1,195 @@
+                }
+
+                if (string.IsNullOrEmpty(content))
+                {
+                    EmbedBuilder b;
+                    b = new EmbedBuilder()
+                    {
+                        Description = $"Usage: `modmail [message]`",
+                        Color = Color.Magenta
+                    };
+                    await Context.Message.Author.SendMessageAsync("", false, b.Build());
+                }
+
+                if (content.Length > 500)
+                {
+                    await Context.Message.Author.SendMessageAsync("Mod mails can not exceed 500 characters");
+                    return;
+                }
+
+                // form embed
+                SocketUser sender = Context.Message.Author;
+                EmbedBuilder builder = new EmbedBuilder()
+                {
+                    Title = "⚠️ | MOD MAIL ALERT!",
+                    Description = $"Modmail from: {sender.Mention} ({sender.Username}#{sender.Discriminator})",
+                    Color = Discord.Color.Gold
+                };
+                builder.WithCurrentTimestamp();
+                builder.AddField("Message Content", $"```{content}```");
+                string messageContent = (role == null) ? "Mod mail" : role.Mention; // role id can be set in database but deleted from server
+                await Context.Channel.SendMessageAsync("Alerting all mods!");
+                await channel.SendMessageAsync(messageContent, false, builder.Build());
+            }
+            catch (Exception ex)
+            {
+                await Context.Channel.SendMessageAsync(ex.ToString());
+                return;
+            }
+        }
+    }
+    [Summary("Modmail configuration commands")]
+    [RequireUserPermission(GuildPermission.Administrator)]
+    [Group("modmailconfig")]
+    public class ModMailConfigModule : InteractiveBase
+    {
+        private FloofDataContext _floofDB;
+
+        public ModMailConfigModule(FloofDataContext floofDB)
+        {
+            _floofDB = floofDB;
+        }
+        private Discord.Color GenerateColor()
+        {
+            return new Discord.Color((uint)new Random().Next(0x1000000));
+        }
+
+        private void CheckServerEntryExists(ulong server)
+        {
+            // checks if server exists in database and adds if not
+            var serverConfig = _floofDB.ModMails.Find(server);
+            if (serverConfig == null)
+            {
+                _floofDB.Add(new ModMail
+                {
+                    ServerId = server,
+                    ChannelId = null,
+                    IsEnabled = false,
+                    ModRoleId = null
+                });
+                _floofDB.SaveChanges();
+            }
+        }
+
+        [Command("channel")]
+        [Summary("Sets the channel for the modmail notifications")]
+        public async Task Channel([Summary("Channel (eg #alerts)")]Discord.IChannel channel)
+        {
```

Comment:
No checks for valid arguments?

Reply:
Set a null check that will set the current channel to the channel to send modmails to